concepts/Vulnerability_disclosure_and_embargo_policy/00004-Vulnerability_disclosure_and_embargo_policy.md

```diff
@@ -0,0 +1,251 @@
+need-to-know people.
+
+In any case, embargoes are a process that should be used sparingly, as they 
+are painful and time-consuming.
+
+For more information about embargoes and vulnerability disclosure, you can 
+refer to [this article](https://aliceevebob.com/2018/01/09/meltdown-and-spectre-thinking-about-embargoes-and-disclosures).
+
+Lastly, in no way should an embargo be understood as a means to avoid making 
+Enarx faults public. Transparency is paramount to the Enarx project.
+
+--- 
+### Enarx Security Advisories
+This page lists all security advisories, vulnerabilities and fixes in Enarx 
+itself.
+
+For security issues, vulnerabilities and fixes in underlying TEEs, both in 
+SDKs and in hardware, please refer to our 
+[page on the subject](https://github.com/enarx/enarx/wiki/TEEs-vulnerabilities-and-attacks/).
+
+If you think you have found a security issue with Enarx, we encourage you to 
+report it.
+To do so, you may follow the process described in the 
+[Vulnerability Disclosure and Embargo Policy](https://github.com/enarx/enarx/wiki/Vulnerability-disclosure-and-embargo-policy) page.
+
+#### List of Enarx Security Advisories
+N/A
+
+--- 
+
+## Reference
+
+Much of this RFC is inspired by the [Rust Security Policy](https://www.rust-lang.org/policies/security),
+as well as [RFPolicy 2.0](https://dl.packetstormsecurity.net/papers/general/rfpolicy-2.0.txt), 
+to which it refers.
+
+## Drawbacks
+
+- This process opts for a "secret by default" approach, as any security issue 
+reported will automatically be in a private area. This may seem at odds with 
+our goal of transparency. We may need a clear set of criteria according to 
+which we make a security report public rather than go for / maintain an embargo.
+
+## Rationale and alternatives
+
+- The most basic alternative is not to have a policy and to simply wait for
+people to disclose their findings. This does nothing to provide guarantees 
+to researchers and users that their work and time will be respected and to 
+encourage further disclosures.  We do not believe that such an approach would be consistent with the transparency and openness goals of the Enarx project.
+- Offer full reporting of security vulnerabilities by email, as many other 
+projects do, and which is what many researchers will be used to.  
+However, encrypted email is not a good solution. The only usable standard for 
+encrypted email is OpenPGP, which suffers from very poor usability, offers no 
+Perfect Forward Secrecy (PFS) and imposes complex and painful key management 
+on the recipients as well as the senders. Cf. [this analysis](https://latacora.micro.blog/2019/07/16/the-pgp-problem.html), 
+for more insights.
+
+## Prior art
+
+Rust process mentioned above.
+
+## Unresolved questions
+Can this process be seen as contradictory with our intention to minimise the 
+use of embargo, and can we make clearer that we don't intend to make every 
+security report secret?
```

Comment:
From what I've seen in cybersecurity, many security researchers are willing to work through private disclosures in order to address the vulnerability as long as the team is responsive. I think this document as a whole presents the team as receptive, so simply following that procedure should work for most researchers. Perhaps publishing an explicit timeline for public disclosure of the details of a vulnerability would reconcile transparency with giving Enarx users time to apply patches to their systems? In order to avoid becoming too prescriptive, this timeline could start after publication of the patch that addresses the vulnerability.

Reply:
Thanks @blazebissar, interesting. I think an explicit timeline for public disclosure might be for a further improvement of this RFC, i'd like to get this one out the door. (And we might need more experience with the process first too.)